websub-ballerina/request_processor.bal

```diff
@@ -0,0 +1,132 @@
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+isolated function processSubscriptionVerification(http:Caller caller, http:Response response, 
+                                                  RequestQueryParams params, SubscriberService subscriberService) {
+    SubscriptionVerification message = {
+        hubMode: params.hubMode,
+        hubTopic: params.hubTopic,
+        hubChallenge: params.hubChallenge,
+        hubLeaseSeconds: params.hubLeaseSeconds
+    };
+
+    SubscriptionVerificationSuccess|SubscriptionVerificationError result = callOnSubscriptionVerificationMethod(subscriberService, message);
+
+    if (result is SubscriptionVerificationError) {
+        result = <SubscriptionVerificationError>result;
+        response.statusCode = http:STATUS_NOT_FOUND;
+        string errorMessage = result.message();
+        response.setTextPayload(errorMessage);
+    } else {
+        response.statusCode = http:STATUS_OK;
+        response.setTextPayload(params.hubChallenge);
+    }
+}
+
+isolated function processSubscriptionDenial(http:Caller caller, http:Response response,
+                                            RequestQueryParams params, SubscriberService subscriberService) {
+    SubscriptionDeniedError subscriptionDeniedMessage = error SubscriptionDeniedError(params.hubReason);
+    
+    var result = callOnSubscriptionDeniedMethod(subscriberService, subscriptionDeniedMessage);
+    
+    if (result is ()) {
+        result = <Acknowledgement>{
+            body: {
+                "message": "Subscription Denial Acknowledged"
+            }
+        };
+    } else {
+        result = <Acknowledgement>result;
+    }
+
+    response.statusCode = http:STATUS_OK;
+    updateResponseBody(response, result["body"], result["headers"]);
+    // respondToRequest(caller, response);
+}
+
+isolated function processEventNotification(http:Caller caller, http:Request request, 
+                                           http:Response response, SubscriberService subscriberService,
+                                           string secretKey) {
+    var payload = request.getTextPayload();
+
+    boolean isVerifiedContent = false;
+    if (payload is string) {
+        isVerifiedContent = verifyContent(request, secretKey, payload);
+    } else {
+        response.statusCode = http:STATUS_INTERNAL_SERVER_ERROR;
+        return;
+    }
+
+    if (!isVerifiedContent) {
+        return;
+    }
+                                               
+    string contentType = request.getContentType();
+
+    map<string|string[]> headers = retrieveRequestHeaders(request);
+
+    ContentDistributionMessage? message = ();
```

Comment:
string contentType = request.getContentType();
map<string|string[]> headers = retrieveRequestHeaders(request);
ContentDistributionMessage? message = ();

I think we can keep those close to each other as they are associated to each other. The stronger that affinity, the less vertical distance there should be between them as explained in Clean Code.

Reply:
* Updated the structure accordingly.